mypy/git.py

```diff
@@ -0,0 +1,119 @@
+"""Utilities for verifying git integrity."""
+
+# Used also from setup.py, so don't pull in anything additional here (like mypy or typing):
+import os
+import pipes
+import subprocess
+import sys
+
+
+def is_git_repo(dir: str) -> bool:
+    """Is the passed directory version controlled with git?"""
+    return os.path.exists(os.path.join(dir, ".git"))
+
+
+def have_git() -> bool:
+    """Can we run the git executable?"""
+    try:
+        subprocess.check_output(["git", "config", "-l"])
```

Comment:
@jhance No; that returns error code `1`, so `check_output` will always raise an error.